gateway/src/apicast/proxy.lua

```diff
@@ -264,15 +293,22 @@ function _M:rewrite(service)
   local var = ngx.var
 
   -- save those tables in context so they can be used in the backend client
-  ctx.usage = usage_params
+  context.usage = context.usage or Usage.new()
+  context.usage:merge(usage)
+
+  ctx.usage = usage
```

Comment:
Now both proxy and the Apicast policy depend on `ctx` and `context`. We'll need to change this to make this cleaner.

Reply:
So `ctx.usage` will be the formatted one and `context.usage` will be our Usage object for now, right?